libbpf-cargo/src/gen/mod.rs

```diff
@@ -524,9 +524,9 @@ fn gen_skel_datasec_getters(
         write!(
             skel,
             r#"
-            pub fn {name}(&mut self) -> &'_ {mutability} {struct_name} {{
+            pub fn {name}(&{mutability} self) -> &{mutability} {struct_name} {{
                 unsafe {{
-                    std::mem::transmute::<*mut std::ffi::c_void, &'_ {mutability} {struct_name}>(
+                    std::mem::transmute::<*mut std::ffi::c_void, &{mutability} {struct_name}>(
```

Comment:
is it intentional that we transmute to *mut c_void for both mutabilities?

Reply:
Not sure which commit you looked at/commented on, but in the end `mutability` would be just be the empty string for the "shared" accessor. So yeah, it's intentional.

Reply:
ah, I commented on commit #2 ("Use non-exclusive receiver for immutable datasec getters"), and it will later be replaced with {ptr_suffix}, never mind.